pkg/kubelet/config/file_test.go

```diff
@@ -125,72 +210,207 @@ func TestReadPodsFromFile(t *testing.T) {
 			}),
 		},
 	}
+}
 
-	for _, testCase := range testCases {
+func (tc *testCase) writeToFile(dir, name string, t *testing.T) *os.File {
+	var versionedPod runtime.Object
+	err := testapi.Default.Converter().Convert(&tc.pod, &versionedPod, nil)
+	if err != nil {
+		t.Fatalf("%s: error in versioning the pod: %v", tc.desc, err)
+	}
+	fileContents, err := runtime.Encode(testapi.Default.Codec(), versionedPod)
+	if err != nil {
+		t.Fatalf("%s: error in encoding the pod: %v", tc.desc, err)
+	}
+
+	fileName := filepath.Join(dir, name)
+	file, err := os.Create(fileName)
+	if err != nil {
+		t.Fatalf("Unable to create test file %#v", err)
+	}
+	file.Close()
+	if err := ioutil.WriteFile(file.Name(), []byte(fileContents), 0555); err != nil {
+		t.Fatalf("Unable to write test file %#v", err)
+	}
+	return file
+}
+
+func watchFileAdded(watchDir bool, t *testing.T) {
+	hostname := "random-test-hostname"
+	var testCases = getTestCases(hostname)
+
+	fileNamePre := "test_pod_config"
+	for index, testCase := range testCases {
 		func() {
-			var versionedPod runtime.Object
-			err := testapi.Default.Converter().Convert(&testCase.pod, &versionedPod, nil)
+			dirName, err := utiltesting.MkTmpdir("dir-test")
 			if err != nil {
-				t.Fatalf("%s: error in versioning the pod: %v", testCase.desc, err)
+				t.Fatalf("Unable to create temp dir: %v", err)
 			}
-			fileContents, err := runtime.Encode(testapi.Default.Codec(), versionedPod)
-			if err != nil {
-				t.Fatalf("%s: error in encoding the pod: %v", testCase.desc, err)
+			defer os.RemoveAll(dirName)
+			fileName := fmt.Sprintf("%s_%d", fileNamePre, index)
+
+			ch := make(chan interface{})
+			if watchDir {
+				NewSourceFile(dirName, hostname, time.Second, ch)
+			} else {
+				NewSourceFile(filepath.Join(dirName, fileName), hostname, time.Second, ch)
 			}
 
-			file := writeTestFile(t, os.TempDir(), "test_pod_config", string(fileContents))
-			defer os.Remove(file.Name())
+			finishEdit := make(chan struct{})
+			changeFile := func() {
+				// Add a file
+				testCase.writeToFile(dirName, fileName, t)
 
-			ch := make(chan interface{})
-			NewSourceFile(file.Name(), hostname, time.Millisecond, ch)
-			select {
-			case got := <-ch:
-				update := got.(kubetypes.PodUpdate)
-				for _, pod := range update.Pods {
-					if errs := validation.ValidatePod(pod); len(errs) > 0 {
-						t.Errorf("%s: Invalid pod %#v, %#v", testCase.desc, pod, errs)
+				if watchDir {
```

Comment:
It is a case to change the file name under the watched dir. It also tests adding a new file under the dir just before this line before which I add a comment `// Add a file`.